streams/core.py

```diff
@@ -602,3 +601,32 @@ def flush(self, _=None):
         out = tuple(self.cache)
         self.emit(out)
         self.cache.clear()
+
+
+def _stream_map(obj, func=None, **kwargs):
+    if hasattr(obj, '__stream_map__'):
```

Comment:
I think I understand better. `singledispatch` serves as a way to add a `__stream_map__` to an object already defined. Useful when the object itself generates new instances of itself. Correct?